### PR TITLE
CORE-6287: Simulating signing service, verification

### PR DESCRIPTION
**Pull request by LizR3**
A key store is passed into each node which will generate a public key on request. Only RSA keys are generated, regardless of spec. The keys are made available for each member through the MemberLookup.

The signing service fakes signing by wrapping the clear data, PEM-encoded key and other parameters in a JSON wrapper.

The verification service parses the wrapper and checks the key, signature spec and clear data against the expected values.

No encryption is ever actually performed. A prominent warning has been added to the README.

### CODE REVIEW

**Comment by mnesbit** [comment]
Note we should always favour ECDSA, or EdDSA over RSA. RSA is not a good statement on our using current cryptography.

**Comment by LizR3** [comment]
@mnesbit Deliberately used RSA to avoid the appearance of actually doing any cryptography (nothing is encrypted!), but I can change it for sure.

File: cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/KeyStore.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package net.corda.cordapptestutils.internal.signing
+
+import net.corda.cordapptestutils.crypto.HsmCategory
+import java.security.PublicKey
+
+interface KeyStore {
```
Comment (alexey-kadyrov-r3) [inline cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/KeyStore.kt]: Don't use `KeyStore` name as it's confusing due that the name is the well defined class in JAVA

Comment (LizR3) [reply cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/KeyStore.kt]: @alexey-kadyrov-r3 Would `SimKeyStore` work, or is that still too similar?

Comment (alexey-kadyrov-r3) [reply cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/KeyStore.kt]: Yep

File: cordapp-test-utils/example-app/src/main/kotlin/net/cordacon/example/AbsenceCallResponderFlow.kt [hunk 2]
```diff
@@ -6,15 +6,24 @@ import net.corda.v5.application.flows.InitiatedBy
 import net.corda.v5.application.flows.ResponderFlow
 import net.corda.v5.application.messaging.FlowSession
 import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.util.contextLogger
 
 @InitiatedBy("absence-call")
 class AbsenceCallResponderFlow: ResponderFlow {
 
+    private companion object {
```
Comment (mnesbit) [inline cordapp-test-utils/example-app/src/main/kotlin/net/cordacon/example/AbsenceCallResponderFlow.kt:14]: We will have to watch the use of companion objects in Corda 5! They are not in fact reliable in the hot-hot HA model that we operate. Logger are special case handled by the platform, but one must never store any state in the companion object

File: cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/BaseSimKeyStore.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    private val keys = HashMap<PublicKey, KeyParameters>()
+    private val keyGenerator = KeyPairGenerator.getInstance("EC")
+
+    init {
+        keyGenerator.initialize(ECGenParameterSpec("secp256k1"), SecureRandom())
+    }
+
+    override fun generateKey(alias: String, hsmCategory: HsmCategory, scheme: String) : PublicKey {
+        val key = keyGenerator.generateKeyPair().public
```
Comment (mnesbit) [inline cordapp-test-utils/test-utils/src/main/kotlin/net/corda/cordapptestutils/internal/signing/BaseSimKeyStore.kt:19]: I am not totally sure why we would throw away the PrivateKey at this point? It seems simplest to my mind to represent the crypto with actual keys and signatures.

**Review by mnesbit — APPROVED** [review]
I will approve so that things move along, but I am not quite sure at some of the design choices here

**Review by ronanbrowne — APPROVED** [review]
LGTM ( Build change reviewed )

**Review by alexey-kadyrov-r3 — APPROVED** [review]
LGTM

**Pull request merged**